lib/grape/validations/validators/multiple_params_base.rb

```diff
@@ -1,26 +1,32 @@
 module Grape
   module Validations
     class MultipleParamsBase < Base
-      attr_reader :scoped_params
-
+      # rubocop:disable HashEachMethods
       def validate!(params)
-        @scoped_params = [@scope.params(params)].flatten
-        params
-      end
+        attributes = AttributesIterator.new(self, @scope, params, multiple_params: true)
+        array_errors = []
 
-      private
+        attributes.each do |resource_params, _|
+          begin
+            validate_params!(resource_params)
+          rescue Grape::Exceptions::Validation => e
+            array_errors << e
+          end
+        end
 
-      def scope_requires_params
-        @scope.required? || scoped_params.any? { |param| param.respond_to?(:any?) && param.any? }
+        raise Grape::Exceptions::ValidationArrayErrors, array_errors if array_errors.any?
       end
+      # rubocop:enable HashEachMethods
```

Comment:
`rubocop -a ; rubocop --auto-gen-config` to prevent Rubocop from being annoying, don't skip this here.